atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -304,8 +304,7 @@ protected void checkGetPreconditions(TableReference tableRef) {
             throw new TransactionFailedRetriableException("Transaction timed out.");
         }
         Preconditions.checkArgument(allowHiddenTableAccess || !AtlasDbConstants.hiddenTables.contains(tableRef));
-        Preconditions.checkState(state.get() == State.UNCOMMITTED || state.get() == State.COMMITTING,
-                "Transaction must be uncommitted.");
+        ensureUncommitted();
```

Comment:
this is a behavior change, though it doesn't seem necessary to support reads on `COMMITTING` transactions.

Reply:
This seems like a good behavior change, though it might affect our big internal product. Especially when serializable transactions are in the mix, we should not allow reads during the commit step.